src/modbus.c

```diff
@@ -281,74 +234,58 @@ int zbx_modbus_read_registers(AGENT_REQUEST *request, AGENT_RESULT *result)
 		return SYSINFO_RET_FAIL;
 	}
 
-	char datatype;
-	int  end = MODBUS_BE_ABCD;	//<endianness> endianness LE(0) BE(1) MBE(2) MLE(3) default BE
-	if (request->nparam > 4)
-	{	// optional params provided
-
-		param5 = get_rparam(request, 4);	// datatype
-
-		if (!strcmp(MODBUS_BIT_STR, param5))
-			datatype = MODBUS_BIT;
-		else if (!strcmp(MODBUS_UINT16_STR, param5))
-			datatype = MODBUS_UINT16;
-		else if (!strcmp(MODBUS_SIGNED_INT_STR, param5))
-			datatype = MODBUS_SIGNED_INT;
-		else if (!strcmp(MODBUS_UINT32_STR, param5))
-			datatype = MODBUS_UINT32;
-		else if (!strcmp(MODBUS_SIGNED_INT32_STR, param5))
-			datatype = MODBUS_SIGNED_INT32;
-		else if (!strcmp(MODBUS_FLOAT_STR, param5))
-			datatype = MODBUS_FLOAT;
-		else if (!strcmp(MODBUS_SIGNED_INT64_STR, param5))
-		{
-			SET_MSG_RESULT(result, strdup("datatype 'int64' is not supported."));
```

Comment:
Removing this breaks ModbusErrorsTestCase.test_int64_datatype 
```
self = <test_modbus_errors.ModbusErrorsTestCase testMethod=test_int64_datatype>
    def test_int64_datatype(self):
        key = "modbus_read_registers["+self.host+",3,14,3,int64]"
>       self.assertEqual(zabbix_get(key),'ZBX_NOTSUPPORTED: datatype \'int64\' is not supported.')
E       AssertionError: '-4629812106207628288.000000' != "ZBX_NOTSUPPORTED: datatype 'int64' is not supported."
tests/test_modbus_errors.py:40: AssertionError
```

Reply:
We were discussing returning a string in this case. This way we don't loose any information on the way from module to Zabbix. There will be inevitable loss on the way from Zabbix to DB, but at least user can decide what's more important - sign or precision.

Reply:
Ok, I changed this test to accept more generic error message.